tokio-util/src/task/task_tracker.rs

```diff
@@ -0,0 +1,629 @@
+    ///
+    /// This is equivalent to `tokio::task::spawn_local(tracker.track_future(task))`.
+    ///
+    /// [`LocalSet`]: tokio::task::LocalSet
+    #[inline]
+    #[track_caller]
+    #[cfg(feature = "rt")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+    pub fn spawn_local<F>(&self, task: F) -> JoinHandle<F::Output>
+    where
+        F: Future + 'static,
+        F::Output: 'static,
+    {
+        tokio::task::spawn_local(self.track_future(task))
+    }
+
+    /// Spawn the provided future on the provided [`LocalSet`], and track it in this `TaskTracker`.
+    ///
+    /// This is equivalent to `local_set.spawn_local(tracker.track_future(task))`.
+    ///
+    /// [`LocalSet`]: tokio::task::LocalSet
+    #[inline]
+    #[track_caller]
+    #[cfg(feature = "rt")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+    pub fn spawn_local_on<F>(&self, task: F, local_set: &LocalSet) -> JoinHandle<F::Output>
+    where
+        F: Future + 'static,
+        F::Output: 'static,
+    {
+        local_set.spawn_local(self.track_future(task))
+    }
+
+    /// Spawn the provided blocking task on the current Tokio runtime, and track it in this `TaskTracker`.
+    ///
+    /// This is equivalent to `tokio::task::spawn_blocking(tracker.track_future(task))`.
+    #[inline]
+    #[track_caller]
+    #[cfg(feature = "rt")]
+    #[cfg(not(target_family = "wasm"))]
+    #[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+    pub fn spawn_blocking<F, T>(&self, task: F) -> JoinHandle<T>
+    where
+        F: FnOnce() -> T,
+        F: Send + 'static,
+        T: Send + 'static,
+    {
+        let token = self.token();
+        tokio::task::spawn_blocking(move || {
+            let res = task();
+            drop(token);
+            res
+        })
+    }
+
+    /// Spawn the provided blocking task on the provided Tokio runtime, and track it in this `TaskTracker`.
+    ///
+    /// This is equivalent to `handle.spawn_blocking(tracker.track_future(task))`.
+    #[inline]
+    #[track_caller]
+    #[cfg(feature = "rt")]
+    #[cfg(not(target_family = "wasm"))]
+    #[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+    pub fn spawn_blocking_on<F, T>(&self, task: F, handle: &Handle) -> JoinHandle<T>
+    where
+        F: FnOnce() -> T,
+        F: Send + 'static,
+        T: Send + 'static,
+    {
+        let token = self.token();
+        handle.spawn_blocking(move || {
+            let res = task();
+            drop(token);
+            res
+        })
+    }
+
+    /// Track the provided future.
+    ///
+    /// The returned [`TrackedFuture`] will count as a task tracked by this collection, and will
+    /// prevent calls to [`wait`] from returning.
+    ///
+    /// The task is removed from the collection when it is dropped. (It is not removed immediately
+    /// when [`poll`] returns [`Poll::Pending`]. You have to actually run the destructor for it to
+    /// be removed.)
```

Comment:
TIOLI, but I might just remove the parenthetical here:
```suggestion
    /// The task is removed from the collection when it is dropped. It is not removed immediately
    /// when [`poll`] returns [`Poll::Pending`]. You have to actually run the destructor for it to
    /// be removed.
```

Reply:
also, did you mean
```suggestion
    /// The task is removed from the collection when it is dropped. (It is not removed immediately
    /// when [`poll`] returns [`Poll::Ready`]. You have to actually run the destructor for it to
    /// be removed.)
```
rather than `Poll::Pending`, here?